Tests/Realm.Tests/Database/RealmDictionaryTests.cs

```diff
@@ -1073,6 +1073,260 @@ public void ObjectFromAnotherRealm_ThrowsRealmException()
+                    Assert.That(insertionSet.InsertedIndices.Length, Is.EqualTo(1));
+                    Assert.That(insertionSet.DeletedIndices, Is.Empty);
+                    Assert.That(insertionSet.ModifiedIndices, Is.Empty);
+
+                    Assert.That(oldDogs.ElementAt(insertionSet.InsertedIndices[0]).Age, Is.EqualTo(i));
+                }
+            }
+
+            _realm.Write(() =>
+            {
+                // Direct insertion
+                dogDict["dog10"] = new Dog { Age = 99 };
+
+                // Indirect insertion: making age fit the iflter
+                dogDict["dog3"].Age = 8;
+
+                // Direct deletion
+                dogDict.Remove("dog9");
+
+                // Indirect deletion: making age no longer fit the filter
+                dogDict["dog8"].Age = 3;
+
+                // Modification
+                dogDict["dog7"].Name = "foo";
+            });
+            _realm.Refresh();
+
+            var changeSet = changeSets.Last();
+            Assert.That(changeSet.InsertedIndices.Length, Is.EqualTo(2));
+            Assert.That(changeSet.DeletedIndices.Length, Is.EqualTo(2));
+            Assert.That(changeSet.ModifiedIndices.Length, Is.EqualTo(1));
+            Assert.That(oldDogs.ElementAt(changeSet.ModifiedIndices[0]).Name, Is.EqualTo("foo"));
+        }
+
+        [Test]
+        public void DictionaryAsRealmQueryable_WhenNotRealmDictionary_Throws()
+        {
+            var dict = new DictionariesObject().ObjectDictionary;
+
+            Assert.That(
+                () => dict.AsRealmQueryable(),
+                Throws.TypeOf<ArgumentException>().And.Message.Contains("dictionary must be an instance of RealmDictionary<IntPropertyObject>"));
+        }
+
+        [Test]
+        public void DictionaryFilter_ReturnsCorrectElementAtResult()
+        {
+            var obj = _realm.Write(() =>
+            {
+                return _realm.Add(new DogDictionaryObject());
+            });
+
+            var dogDict = obj.Dict;
+            var oldDogs = dogDict.Filter("Age >= 5").OrderBy(d => d.Age);
+
+            _realm.Write(() =>
+            {
+                for (var i = 0; i < 10; i++)
+                {
+                    dogDict.Add("dog" + i, new Dog { Age = i });
+                }
+            });
+
+            for (var i = 0; i < 5; i++)
+            {
+                var dog = oldDogs.ElementAt(i);
+                Assert.That(dog.Age, Is.EqualTo(i + 5));
+            }
+        }
+
+        [Test]
+        public void DictionaryFilter_PassesArgumentsCorrectly()
+        {
+            var obj = _realm.Write(() =>
+            {
+                return _realm.Add(new DogDictionaryObject());
+            });
+
+            var dogDict = obj.Dict;
+            var fiDogs = dogDict.Filter("Name BEGINSWITH[c] $0", "Fi");
+
+            _realm.Write(() =>
+            {
+                dogDict.Add("Ris", new Dog { Name = "Rick" });
```

Comment:
I am using different keys just to make sure there's no overlap between keys and values in functionality